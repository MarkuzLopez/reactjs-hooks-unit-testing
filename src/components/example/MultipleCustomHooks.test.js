import React from 'react';
import { shallow } from "enzyme";
import MultipleCustomHooks from "./MultipleCustomHooks";
import { useFetch } from '../../hooks/useFetch';
import { useCounter } from '../../hooks/useCounter';

jest.mock('../../hooks/useFetch.js');
// jest.mock('../../hooks/useCounter.js');


describe('Pruebas en <MultipleCustomHooks />', () => {

    // test('should ', () => {
    //     useCounter.mockReturnValue({
    //         counter: 10,
    //         increment: () => { }
    //     });
        
    // })
    

    test('debe de mostrarse correctamente ', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar la informacion', () => {

        useFetch.mockReturnValue({ 
            data: [{
                author: 'Markuz',
                quote: 'Hola probando'
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks />);
        expect( wrapper.find('.alert').exists() ).toBe(false);
        expect( wrapper.find('.mb0').text().trim() ).toBe('Hola probando');
        expect( wrapper.find('footer').text().trim()).toBe('Markuz')
        // expect(wrapper).toMatchSnapshot();

    })
    
    
})
