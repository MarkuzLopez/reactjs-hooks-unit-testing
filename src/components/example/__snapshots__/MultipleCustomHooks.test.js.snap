// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks /> debe de mostrarse correctamente  1`] = `
<Fragment>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente
  </button>
</Fragment>
`;
